### PR TITLE
"Is Order shippable" icon

### PR DESCRIPTION
**Pull request by frederic34**

### CODE REVIEW

File: htdocs/commande/list.php [hunk 1]
```diff
@@ -325,22 +326,54 @@
...
+            $shippable=0;
+            $text_info='';
+            for ($lig=0; $lig<(count($generic_commande->lines)); $lig++) {
+                if ($generic_commande->lines[$lig]->product_type==0) {
+                    $text_info .= $generic_commande->lines[$lig]->qty.' X '.$generic_commande->lines[$lig]->ref.' : '.dol_trunc($generic_commande->lines[$lig]->product_label, 28);
+                    $generic_product->id = $generic_commande->lines[$lig]->fk_product;
+                    $generic_product->load_stock();
+                    if ($generic_product->stock_reel<$generic_commande->lines[$lig]->qty) {
+                        $shippable++;
+                        $text_info.='<span style="color:#FF0000">'.$langs->trans('Available').'&nbsp;:&nbsp;'.$generic_product->stock_reel."</span><br>";
```
Comment (hregis) [inline htdocs/commande/list.php]: use css file please, no "style" in the code
thanks

**Comment by frederic34** [comment]
shipicon

**Comment by atm-maxime** [comment]
Subject interest me because we worked on this (shippable order module).
The result is ok if you consider each order separately. But like this, in the list all orders can appear as "shippable" when really after shipping the first one, some may become "unshippable"...
And it bothers me a bit to see "shippable++" when stock is not enough and "shippable == 0" to display "Shippable".

**Comment by frederic34** [comment]
shippable

**Comment by frederic34** [comment]
You can deactivate in case of use of a better module or if it slow down your list.

**Comment by atm-maxime** [comment]
Ok, so my comment was useless I guess...
I maintain this dev is wrong. For example I have 2 orders containing the same product, it will appear in the list that the 2 orders are shippable, but in reality, when the first one will be shipped, I won't be able to ship the 2nd one...

c1410001
c1410002

**Comment by frederic34** [comment]
Your comments are not useless, that's why I need to improve computation of "stock theorique" for all case of stock increase/decrease and put an alert if there isn't enough products for all orders.

**Comment by BebZ** [comment]
Thnaks Frederic for your first commit, the second one, as maxime requests will be very useful and more "realistic" to get the "sum of products to ship on validated orders" - "products already shipped on validated or on shipping orders" compared to stock physique.
Stock thorique is a little confusing for the reason it counts ALL the products on validated orders, and doens't decrease this amount of "already shipped products"... but I'm sure you are quite near to find the solution :)

**Comment by atm-maxime** [comment]
Hi Bebz,
FYI, in 3.7 I have corrected the virtual stock calculation to have something like :
Virtual Stock = Real stock - (customer orders - customer shipments) + (supplier orders - supplier receptions)
That's of course when you work with stock calculation on shipments. See here :
https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/product/class/product.class.php#L3035

**Comment by BebZ** [comment]
thanks maxime, i think frederic34 did also a commit in this direction, no ?
https://github.com/Dolibarr/dolibarr/commit/f23f16da861d128c7f1e2e355fbfd151a0c7b4ac

Whet's the diffrence, as an end user whent testing ? or is it complementary ?

I tested the new configuration of order shippable it is really practical, and also the stock's product card infos of already delivered/ordered products...

**Pull request merged**